src/schema/clan/setRole.ts

```diff
@@ -0,0 +1,24 @@
+import { Type } from "@sinclair/typebox";
+
+import { defineEndpoint } from "@/generator-helpers.js";
+
+export default defineEndpoint({
+    source: "user",
+    target: "server",
+    description: "Set the target player's role in your clan.",
+    request: {
+        data: Type.Object({
+            userId: Type.Ref("userId"),
+            clanId: Type.Ref("clanId"),
+            targetRole: Type.Ref("clanRole"),
+        }),
+    },
+    response: [
+        { status: "success" },
+        { status: "failed", reason: "only_one_leader" },
```

Comment:
`only_one_leader` cannot happen. Because if the leader set the leader role to someone else, they transactionally lose their leader role, and so the invariant of "only one leader" is held.